frontend/src/libs/components/loader/loader.tsx

```diff
@@ -6,18 +9,39 @@ type Properties = {
   children: React.ReactNode;
   className?: string;
   isLoading: boolean;
+  hasOverlay: boolean;
+  loaderType: ValueOf<typeof ILoader>;
 };
 
-const Loader: React.FC<Properties> = ({ children, className, isLoading }) => {
-  if (!isLoading) {
-    return children;
+const Loader: React.FC<Properties> = ({
+  children,
+  className,
+  isLoading,
+  hasOverlay,
+  loaderType,
+}) => {
+  const loaderComponents: Record<string, React.ReactNode> = {
+    'circular': (
+      <div
+        className={getValidClassNames(styles.circularLoader, className)}
+      ></div>
+    ),
+    'dots': (
+      <div className={getValidClassNames(styles.dotFlashing, className)}></div>
+    ),
+  };
```

Comment:
The only thing different in this map is className passed - just pass it by condition